pf9/support/generate_bundle.py

```diff
@@ -0,0 +1,199 @@
+            if host in ['127.0.0.1','localhost']:
+                hostname=socket.gethostname()
+                ip=socket.gethostbyname(hostname)
+            # get host uuid from resmgr using the ip of the localhost
+                host_uuid[ip]=self.get_uuid_from_resmgr(du_url,ip,headers)
+
+                if host_uuid[ip]:
+                # if a valid uuid is found , hostagent is configured and we can pull the support bundle from the host
+                    resmgr_endpoint = '{}/resmgr/v1/hosts/{}'.format(ctx.params['du_url'],host_uuid[ip])
+                    resmgr_get_hosts = requests.get(resmgr_endpoint,verify=False,headers=headers)
+                    data = resmgr_get_hosts.json()
+                    roles = (data["roles"])
+                    if len(roles) >= 1 :
+                        if roles[0] == "pf9-kube" and data["info"]["responding"] == "true":  # Responding key will only be there is pf9-kube is installed else there will not be any responding key in response
+                            pass
+                        else:
+                            time.sleep(2)
+                            self.create_log_bundle(ctx,user,password,host)
+                    else:
+                        time.sleep(2)
+                        self.create_log_bundle(ctx,user,password,host)
+                else:
+                    self.upload_pf9cli_logs (du_url,host)
+
+            # host provided in not localhost, get host uuid from resmgr using the ip
+            else:
+                host_uuid[host]=self.get_uuid_from_resmgr(du_url,host,headers)
+                if host_uuid[host]:
+                    resmgr_endpoint = '{}/resmgr/v1/hosts/{}'.format(ctx.params['du_url'],host_uuid[host])
+                    resmgr_get_hosts = requests.get(resmgr_endpoint,verify=False,headers=headers)
+                    data = resmgr_get_hosts.json()
+                    roles = (data["roles"])
+                    if len(roles) >= 1:
+                        if roles[0] == "pf9-kube" and data["info"]["responding"] == "true": # Responding key will only be there is pf9-kube is installed else there will not be any responding key in response
+                            pass
+                        else:
+                            self.create_log_bundle(ctx,user,password,host)
+                    else:
+                        self.create_log_bundle(ctx,user,password,host)
+                else:
+                # if a valid uuid not found in resmgr from the host , hostagent is not installed and support bundle cannot be generated
+                # uploading pf9ctl logs in this case
+                    self.upload_pf9cli_logs (du_url,host)
+
+        return None
+
+    def upload_pf9cli_logs (self,du_url,host):
+    #function for uploading pf9cli logs in case support bundle cant be generated
+        log_path = str(self.error_msg)
+        filename = log_path.split(":")[2].lstrip()
+        cli_logs="/root/pf9/log/pf9ctl.log"
```

Comment:
This won't capture the ansible logs in this dir? Can we just upload all files in this dir? I think the ansible logs start with node_provision...

Reply:
Paavan, The prep_node() function returns the log path for the ansible logs(node_provision) we are using the same log path to upload the current ansible logs to which prep_node() was logging instead of uploading the entire directory.